ts/components/session/SessionSeedModal.tsx

```diff
@@ -0,0 +1,215 @@
+    this.getSeed = this.getSeed.bind(this);
+    this.confirmPassword = this.confirmPassword.bind(this);
+    this.checkHasPassword = this.checkHasPassword.bind(this);
+    this.onEnter = this.onEnter.bind(this);
+
+    window.addEventListener('keyup', this.onEnter);
+  }
+
+  public render() {
+    const i18n = window.i18n;
+    const { onClose } = this.props;
+
+    const maxPasswordLen = 64;
+
+    this.checkHasPassword();
+    void this.getSeed();
+
+    const error = this.state.error;
+    const hasPassword = this.state.hasPassword;
+    const passwordValid = this.state.passwordValid;
+
+    const loading = this.state.loadingPassword || this.state.loadingSeed;
+
+    return (
+      <>
+        {!loading && (
+          <SessionModal
+            title={i18n('showSeed')}
+            onOk={() => null}
+            onClose={onClose}
+          >
+            <div className="spacer-sm" />
+
+            {hasPassword && !passwordValid ? (
+              <div>
+                <p>{i18n('showSeedPasswordRequest')}</p>
+                <input
+                  type="password"
+                  id="seed-input-password"
+                  placeholder={i18n('password')}
+                  maxLength={maxPasswordLen}
+                />
+
+                {error && (
+                  <>
+                    <div className="spacer-xs" />
+                    <div className="session-label danger">{error}</div>
+                  </>
+                )}
+
+                <div className="spacer-lg" />
+
+                <div className="session-modal__button-group">
+                  <SessionButton
+                    text={i18n('confirm')}
+                    onClick={this.confirmPassword}
+                  />
+
+                  <SessionButton text={i18n('cancel')} onClick={onClose} />
+                </div>
+              </div>
+            ) : (
+              <>
+                <div className="session-modal__centered text-center">
+                  <p className="session-modal__description">
+                    {i18n('seedSavePromptMain')}
+                    <br />
+                    <span className="text-subtle">
+                      {i18n('seedSavePromptAlt')}
+                    </span>
+                  </p>
+                  <div className="spacer-xs" />
+
+                  <i className="session-modal__text-highlight">
+                    {this.state.seed}
+                  </i>
+                </div>
+                <div className="spacer-lg" />
+
+                <div className="session-modal__button-group">
+                  <SessionButton text={i18n('ok')} onClick={onClose} />
+
+                  <SessionButton
+                    text={i18n('copySeed')}
+                    onClick={() => {
+                      this.copySeed(this.state.seed);
+                    }}
+                  />
+                </div>
+              </>
+            )}
+          </SessionModal>
+        )}
```

Comment:
make all those terniary operator to sub render() methods rathert than here. it will be easier to read.